tls/s2n_psk.c

```diff
@@ -599,3 +599,39 @@ int s2n_connection_set_psk_mode(struct s2n_connection *conn, s2n_psk_mode mode)
     conn->psk_mode_overridden = true;
     return S2N_SUCCESS;
 }
+
+int s2n_connection_get_negotiated_psk_identity_length(struct s2n_connection *conn, uint16_t *identity_length)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(identity_length);
+
+    struct s2n_psk *chosen_psk = conn->psk_params.chosen_psk;
+
+    if (chosen_psk == NULL) {
+        POSIX_BAIL(S2N_ERR_NEGOTIATED_PSK_NOT_FOUND);
+    }
+
+    *identity_length = chosen_psk->identity.size;
+
+    return S2N_SUCCESS;
+}
+
+int s2n_connection_get_negotiated_psk_identity(struct s2n_connection *conn, uint8_t *identity,
+                                               uint16_t *identity_length)
```

Comment:
You actually need "max_identity_length", not "identity_length". Just a value, not a pointer. There's a separate API to get the actual length, what you need to know is how large "identity" is.

Reply:
This parameter is both an input and output one. As the input parameter, `identity_length` represents the`max_identity_length` for the identity and as an output parameter it would be set to the actual length of the `identity` to convey to the customer what the actual identity value is here.